### PR TITLE
feat(nuxt): init, app, component, page generators

### PR DESCRIPTION
**Pull request by mandarini**
* `@nx/nuxt:init`
* `@nx/nuxt:app`
* `@nx/nuxt:component`
* `@nx/nuxt:page`

generators

### CODE REVIEW

**Review by jaysoo — APPROVED** [review]

File: e2e/nuxt/src/nuxt.test.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import {
+  cleanupProject,
+  killPorts,
+  newProject,
+  runCLI,
+  uniq,
+} from '@nx/e2e/utils';
+
+describe('Vue Plugin', () => {
```
Comment (JamesHenry) [inline e2e/nuxt/src/nuxt.test.ts:9]: Nuxt right?

File: e2e/nuxt/src/nuxt.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  let proj: string;
+  const app = uniq('app');
+
+  beforeAll(() => {
+    proj = newProject({
+      unsetProjectNameAndRootFormat: false,
+    });
+    runCLI(
+      `generate @nx/vue:app ${app} --unitTestRunner=vitest --e2eTestRunner=playwright`
```
Comment (JamesHenry) [inline e2e/nuxt/src/nuxt.test.ts:18]: What purpose does this vue app serve?

File: packages/nuxt/src/generators/init/init.spec.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { nuxtInitGenerator } from './init';
+
+describe('init', () => {
+  let tree: Tree;
+
+  beforeEach(() => {
+    tree = createTreeWithEmptyWorkspace();
+  });
+
+  it('should add vue dependencies', async () => {
```
Comment (JamesHenry) [inline packages/nuxt/src/generators/init/init.spec.ts:12]: This one is a nit, but given other changes likely need to be made...

```suggestion
  it('should add vue and nuxt dependencies', async () => {
```

File: packages/nuxt/src/generators/page/page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
+import { Tree } from '@nx/devkit';
+import { applicationGenerator } from '../application/application';
+import { getDirectory, pageGenerator } from './page';
+
+describe('app', () => {
```
Comment (JamesHenry) [inline packages/nuxt/src/generators/page/page.spec.ts:6]:
```suggestion
describe('page', () => {
```

**Review by JamesHenry — CHANGES_REQUESTED** [review]

File: packages/nuxt/package.json [hunk 5]
```diff
@@ -28,7 +28,17 @@
...
+    "defu": "^6.1.2",
+    "fs-extra": "^11.1.0",
+    "tslib": "^2.3.0",
+    "@nx/devkit": "file:../devkit",
+    "@nx/js": "file:../js",
+    "@nx/linter": "file:../linter",
+    "@nx/vue": "file:../vue",
+    "@nx/cypress": "file:../cypress",
+    "@nx/playwright": "file:../playwright",
+    "@nuxt/kit": "^3.7.4",
```
Comment (jaysoo) [inline packages/nuxt/package.json:40]: We should install `@nuxt/*` packages in the init generator. This way users can update their versions without any conflicts with ours. But we can add a `peerDependencies` entry for `@nuxt/kit` at version `>= 3.0.0`.

**Review by JamesHenry — APPROVED** [review]

**Pull request merged**